processing/src/main/java/io/druid/query/aggregation/AggregationTestHelper.java

```diff
@@ -0,0 +1,319 @@
+      while (rows.hasNext()) {
+
+        Object row = rows.next();
+        if (row instanceof String && parser instanceof StringInputRowParser) {
+          //Note: this is required because StringInputRowParser is InputRowParser<ByteBuffer> as opposed to
+          //InputRowsParser<String>
+          index.add(((StringInputRowParser) parser).parse((String) row));
+        } else {
+          index.add(parser.parse(row));
+        }
+      }
+      IndexMerger.persist(index, outDir, new IndexSpec());
+    }
+  }
+
+  //Simulates running group-by query on individual segments as historicals would do, json serialize the results
+  //from each segment, later deserialize and merge and finally return the results
+  public Sequence<Row> runQueryOnSegments(final List<File> segmentDirs, final String groupByQueryJson) throws Exception
+  {
+    return runQueryOnSegments(segmentDirs, mapper.readValue(groupByQueryJson, GroupByQuery.class));
+  }
+
+  public Sequence<Row> runQueryOnSegments(final List<File> segmentDirs, final GroupByQuery query)
+  {
+    final List<QueryableIndexSegment> segments = Lists.transform(
+        segmentDirs,
+        new Function<File, QueryableIndexSegment>()
+        {
+          @Override
+          public QueryableIndexSegment apply(File segmentDir)
+          {
+            try {
+              return new QueryableIndexSegment("", IndexIO.loadIndex(segmentDir));
+            }
+            catch (IOException ex) {
+              Throwables.propagate(ex);
+            }
+            throw new RuntimeException("unreachable");
+          }
+        }
+    );
+
+    try {
+      final FinalizeResultsQueryRunner baseRunner = new FinalizeResultsQueryRunner(
+          toolChest.postMergeQueryDecoration(
+              toolChest.mergeResults(
+                  toolChest.preMergeQueryDecoration(
+                      new ConcatQueryRunner(
+                          Sequences.simple(
+                              Lists.transform(
+                                  segments,
+                                  new Function<Segment, QueryRunner>()
+                                  {
+                                    @Override
+                                    public QueryRunner apply(final Segment segment)
+                                    {
+                                      try {
+                                        return makeStringSerdeQueryRunner(
+                                            mapper,
+                                            toolChest,
+                                            query,
+                                            factory.createRunner(segment)
+                                        );
+                                      }
+                                      catch (Exception ex) {
+                                        Throwables.propagate(ex);
```

Comment:
you can simply throw the result of Throwables.propagate to make the compiler happy